WalletWasabi.Fluent.Desktop/Program.cs

```diff
@@ -12,38 +12,26 @@
 using WalletWasabi.Fluent.CrashReport;
 using WalletWasabi.Fluent.Helpers;
 using WalletWasabi.Fluent.ViewModels;
-using WalletWasabi.Helpers;
 using WalletWasabi.Logging;
 using WalletWasabi.Models;
-using WalletWasabi.Services;
-using WalletWasabi.Services.Terminate;
-using WalletWasabi.Wallets;
 using System.Diagnostics.CodeAnalysis;
 using WalletWasabi.Fluent.Desktop.Extensions;
 using System.Net.Sockets;
 using System.Collections.ObjectModel;
+using WalletWasabi.Daemon;
 using LogLevel = WalletWasabi.Logging.LogLevel;
 
 namespace WalletWasabi.Fluent.Desktop;
 
 public class Program
 {
-	private static Global? Global;
-
 	// Initialization code. Don't use any Avalonia, third-party APIs or any
 	// SynchronizationContext-reliant code before AppMain is called: things aren't initialized
 	// yet and stuff might break.
-	public static int Main(string[] args)
+	public static async Task<int> Main(string[] args)
 	{
-		bool runGuiInBackground = args.Any(arg => arg.Contains(StartupHelper.SilentArgument));
-
-		// Initialize the logger.
-		string dataDir = EnvironmentHelpers.GetDataDir(Path.Combine("WalletWasabi", "Client"));
-		SetupLogger(dataDir, args);
-
-		Logger.LogDebug($"Wasabi was started with these argument(s): {(args.Any() ? string.Join(" ", args) : "none")}.");
-
 		// Crash reporting must be before the "single instance checking".
+		Logger.InitializeDefaults(Path.Combine(Config.DataDir, "Logs.txt"), LogLevel.Info);
```

Comment:
Yes. However the Desktop consists of two mutually-exclusive apps: CrashReport and Desktop. The problem is CrashReport uses the Logger and then it has to be initialized (sucks). However, reinitializing  the logger doesn't create problems.

IMO the CrashReport should be a different app instead of a hack like this but i think the price that we paid for such a ugliness is not big.